plugin/pkg/scheduler/algorithm/scheduler_interface.go

```diff
@@ -18,10 +18,25 @@ package algorithm
 
 import (
 	"k8s.io/kubernetes/pkg/api"
+	schedulerapi "k8s.io/kubernetes/plugin/pkg/scheduler/api"
 )
 
+// SchedulerExtender is an abstract interface for external processes to influence scheduling
+// decisions made by Kubernetes.
+type SchedulerExtender interface {
+	// Filter based on extender implemented predicate functions. The filtered list is
+	// expected to be a subset of the supplied list.
+	Filter(pod *api.Pod, nodes *api.NodeList) (*api.NodeList, error)
+
+	// Prioritize based on extender implemented priority functions. Weight*priority is added
+	// up for each such priority function. The returned score & weight are used to compute
+	// the weighted score for an extender. The weighted score is added to the score computed
+	// by Kubernetes scheduler. The total score is used to do the host selection.
+	Prioritize(pod *api.Pod, nodes *api.NodeList) (*schedulerapi.HostPriorityList, int, error)
```

Comment:
I don't have a strong opinion, though "prioritize" seems slightly better since we call them "priority functions" so it sounds a little more consistent.